### PR TITLE
Desktop: set cancel as the default in dangerous operations

### LINKED ISSUE #1662: UI issues (delete notebook/tags/notes)
### Operating system

- macOS

### Application

- Desktop

---

I've opened only one issue,  because they are all related.
I have noticed problems with the dialog boxes for the following actions:

**Delete notebook** (right click on a notebook, select Delete)

`OK` is selected by default. This is dangerous.
It is not possible to use the keyboard to switch between `Cancel` and `OK`.
`OK` button should be renamed to `Delete` or `DELETE`.
`Cancel` should be selected by default.

**Delete tags** (right click on a tag, select Delete)

The context menu item should be renamed from `Delete` to `Remove`. We are not deleting the tag.

`OK` is selected by default. This is dangerous.
It is not possible to use the keyboard to switch between `Cancel` and `OK`.
`OK` button should be renamed to `Remove` or `REMOVE`.
`Cancel` should be selected by default.

**Delete a single note or multiple notes**

No button is selected by default. It is not possible to use the keyboard to select `Cancel` or `Delete`.
`Cancel` should be selected by default.
If above dialog boxes were to use UPPERCASE for remove/delete, so should this one.

image

@laurent22 I remember you added a change a while back, so I am not sure why this is happening.

### PR DESCRIPTION
**Pull request by tessus**
and rename buttons (delete, remove) and context menu (for tags)

fixes #1662 (partially, because I can't fix the upstream issue, unless we update Electron)

Please note that the reason for `Cancel` not being selected is an upstream issue. But this was already the case for the `delete note dialog` (see below).
Therefore I made everything more consistent.

## right click on tag

image

## remove tag dialog

image

## delete notebook dialog

image

## delete note dialog (did not change anything, only listed as a reference)

image

### CODE REVIEW

File: ElectronClient/app/gui/SideBar.jsx [hunk 1]
```diff
@@ -267,14 +267,18 @@ class SideBarComponent extends React.Component {
 		if (!itemId || !itemType) throw new Error('No data on element');
 
 		let deleteMessage = '';
+		let buttonLabel = '';
```
Comment (laurent22) [inline ElectronClient/app/gui/SideBar.jsx]: Could you give it `_('Remove')` as a default value please? Then you can remove the two `buttonLabel = _('Remove');` below. That way, if there's ever another condition added below, the button label would still have a good default.

**Comment by laurent22** [comment]
That looks good other than the minor comment I've added.

File: ElectronClient/app/gui/SideBar.jsx (hunk 1 above)
Comment (tessus) [reply ElectronClient/app/gui/SideBar.jsx]: Sure, will do.

**Comment by tessus** [comment]
Done. The reason why I didn't do it in the first plac was because I'm very careful with default values and fall through logic. If someone later one changes the value, it could mess up the flow (and output).

**Comment by laurent22** [comment]
Perfect, thanks for the update!

**Pull request merged**